blaseball_mike/chronicler.py

```diff
@@ -133,3 +133,23 @@ def get_tribute_updates(before=None, after=None, order=None, count=None):
         params["count"] = count
 
     return paged_get(f'{BASE_URL}/tributes/hourly', params=params, session=cached_session)
+
+def time_map(season=0, tournament=-1, day=0, include_nongame=False):
+    # Return start & end timestamps for a given gameday
+    season = season - 1
+    day = day - 1
+
+    map = cached_session.get(f'{BASE_URL}/time/map').json()
+
+    # Filter out desired events
+    if tournament != -1:
+        # Season is not always -1 if a tournament is active, so ignore it
+        results = list(filter(lambda x: x['tournament'] == tournament and x['day'] == day, map['data']))
+    else:
+        results = list(filter(lambda x: x['tournament'] == -1 and x['season'] == season and x['day'] == day, map['data']))
+
+    # Optionally filter out phase-change events
+    if not include_nongame:
+        results = list(filter(lambda x: x['type'] in ('season', 'tournament', 'postseason'), results))
+
+    return results
```

Comment:
can `results` be guaranteed to be a `datetime` object?

Reply:
This currently returns a list of dicts of the data from chronicler, in the following format
```
[
  {
    "season":1,
    "tournament":-1,
    "day":98,
    "type":"season",
    "startTime":"2020-08-01T07:13:21.108Z",
    "endTime":"2020-08-01T13:00:02.705Z"
  }
]
```
I'm of a mixed mind on this, since the user really only needs the start & end times and the season/day are 0-indexed unlike the rest of mike. I could remove everything but the start/end times from here but you'd probably still want `type` in case the user requested the pre-election/post-election phase timestamps as well.

Reply:
i think this format is actually fine, i just wanted to make sure the documentation captures this. Two modifications, though: 1) pop the dict out of the list, seems unnecessary, 2) convert those strings into proper datetimes.

Reply:
1 is tricky, since chronicler has separate elements for each sim phase but they keep the same season/day. For example, looking for Season 11 Day 120 returns 3 dicts with types of `postseason`, `pre-election`, and `post-election`. These are currently only included if the `include_nongame` flag is set, otherwise it's just the postseason one.

I could also just always exclude the non-game phases and remove the flag, not sure if people are really going to want to get that granular with it.

Reply:
ok that makes sense. keep it as is, and we can clean it up in the model, or maybe util or something

Reply:
Alright, output times are now datetimes